test/index.test.ts

```diff
@@ -613,385 +613,1055 @@ describe('HipThrusTS', () => {
 
         const triple = HTPipe(left, midNotCovered, rightFullyCovered);
       });
-      it('attaches properly typed data from left and right sync data attacher', async () => {
-        const aPassedIn = 'some string';
-        const bPassedIn = 4;
-        const cReturned = 6;
-        const left = {
-          attachData(context: { a: string }) {
-            expect(context.a).to.be.equal(aPassedIn);
-            return { b: bPassedIn };
-          },
-        };
 
-        const rightFullyCovered = {
-          attachData(context: { b: number }) {
-            expect(context.b).to.be.equal(bPassedIn);
-            return { c: cReturned };
-          },
-        };
+      const basicPipedCaseTestConstants = {
+        aPassedIn: 'some string',
+        bPassedIn: 4,
+        cReturned: 6,
+      };
 
-        const pipedAtoBC = HTPipe(left, rightFullyCovered);
+      const basicPipedCaseTestSyncReturns = {
+        leftProjectorReturn: { b: basicPipedCaseTestConstants.bPassedIn },
+        rightProjectorReturn: { c: basicPipedCaseTestConstants.cReturned },
+      };
 
-        await HTPipeTest(
-          pipedAtoBC,
-          'attachData',
-          { a: aPassedIn },
-          { b: bPassedIn, c: cReturned },
-          true
-        );
-      });
-      it('attaches right sync output instead of left sync output if right output sync transform left output type', async () => {
-        const aPassedIn = 5;
-        const bPassedIn = 'some string';
-        const bReturned = 4;
-
-        const left = {
-          attachData(context: { a: number }) {
-            expect(context.a).to.be.equal(aPassedIn);
-            return { b: bPassedIn };
-          },
+      const basicPipedCaseTestAsyncReturns = {
```

Comment:
all this changes is needed for lifecycles what can return only sync
when we have a condition to choose we returning sync or async we will have a type mismatch in HTPipeTest function